File: python_implementation/src/application/use_cases/tasks/delete_task_use_case.py
from uuid import UUID
from src.domain.abstractions.repositories.abstract_task_repository import AbstractTaskRepository


class DeleteTaskUseCase:
    """Use case for deleting a task"""
    
    def __init__(self, task_repository: AbstractTaskRepository):
        self.task_repository = task_repository
    
    def execute(self, task_id: UUID) -> bool:
        return self.task_repository.delete(task_id)

File: java_implementation/src/main/java/com/example/taskapi/application/usecases/DeleteTaskUseCase.java
package com.example.taskapi.application.usecases;

import com.example.taskapi.domain.repository.TaskRepositoryPort;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.UUID;

@Service
@RequiredArgsConstructor
public class DeleteTaskUseCase {
    private final TaskRepositoryPort taskRepository;

    public boolean execute(UUID taskId) {
        return taskRepository.delete(taskId);
    }
}
